plutus-scb/src/Plutus/SCB/Core/ContractInstance.hs

```diff
@@ -172,6 +179,23 @@ processAllContractInboxes = do
     itraverse_ (processFirstInboxMessage @t) state
     logInfo "processAllContractInboxes end"
 
+-- | Check the inboxes of all contracts.
+processContractInbox ::
+    forall t effs.
+        ( Member (EventLogEffect (ChainEvent t)) effs
+        , Member (ContractEffect t) effs
+        , Member (Error SCBError) effs
+        , Member Log effs
+        )
+    => ContractInstanceId
+    -> Eff effs ()
+processContractInbox i = do
+    logInfo "processContractInbox start"
+    logInfo $ render $ pretty i
+    state <- runGlobalQuery (Query.inboxMessages @t)
+    itraverse_ (processFirstInboxMessage @t) (Map.filterWithKey  (\k _ -> k == i) state)
```

Comment:
This seems an odd way to express it, when `filterWithKey (\k _ -> k == i)` is essentially `Map.lookup`. :puzzled-face:

Reply:
Good point :)